app/services/ui_notifications/hosts/provision.rb

```diff
@@ -0,0 +1,51 @@
+module UiNotifications
+  module Hosts
+    class Provision < Base
+      def deliver!
+        case state
+        when :build_completed
+          host_build
+        end
+      end
+
+      private
+
+      def host_build
+        logger.debug('Host finished provisioning (build is done) - checking for notficiations')
+        blueprint = host_build_blueprint
+
+        if blueprint.notifications.where(:subject => subject).empty?
+          logger.debug('creating new UI notification host_build_completed')
+          ::Notification.create!(
+            initiator: User.find(subject.owner_id),
+            subject: subject,
+            # TODO fix usrgroup
+            audience: audience,
+            notification_blueprint: blueprint
+          )
+        else
+          logger.debug('UI notification already exists, updating timestamps.')
+          blueprint.mass_update_expiry
+        end
+      end
+
+      def allowed_states
+        [:build_completed]
+      end
+
+      def audience
+        if subject.owner_type == 'User'
+          ::Notification::AUDIENCE_USER
+        else
```

Comment:
If the host has no owner (very common case), there is no notification? Taxonomy could fit in that case